Throw a warning when writing code with invalid urls

jjwon opened the pull request:
Deals with #306.  Technically, this only supports urls which are "invalid" because they are missing a protocol, and therefore not linking to the expected place.

I wanted to verify two things:
- I modified Slowparse for this to work &mdash; it felt like the least-obstructive way to do that.  Is this alright?  I was a little bothered because Slowparse is more of an HTML validator, and technically there is nothing invalid about links with missing protocols.
- I throw a `ParseError` when I find a problem.  I noticed that there was [code](https://github.com/jjwon/live-editor/blob/b5f9192567b64ecd3674013535db0fd8170ed11e/external/slowparse/slowparse.js#L1746) that added to some `warnings` array, but I couldn't figure out how this got used.

Also, a lot of trailing whitespaces got eaten by my editor, which is why so many lines are changed.

*Maybe To Do?*
- [ ] Add a new error when urls are actually invalid, and replace the invalid symbols by their corresponding percent-encodings (it seems that this is already handled by the link-redirector, though)

kevinb7 commented:
@jjwon Thanks for the pull request.  I'll try to have a look it this weekend.  Could you sign our Contributor License Agreement: http://www.khanacademy.org/r/cla?

jjwon commented:
Signed!  Thanks.

kevinb7 commented:
I wouldn't worry about the trailing whitespace being eaten.  I recently added .editorconfig to eat trailing whitespace but haven't fixed all of the files so this will be expected for the next little while.

kevinb7 commented:
@jjwon thanks for signing the CLA.  I had a look at the code.  Checking that "www" is in the attribute value will miss cases where the host name doesn't start "www".  It's probably better to check for `attributeName === "href"` and `attributeName === "src"`.  Thanks for adding tests.  Please also add a failing tests for `<a href='google.com'></a>`.

jjwon commented:
@kevinb7 I updated the code using the `attributeName`, and also using a general invalid URL regex instead of just the protocol matching.  I changed error names accordingly.
